### PR TITLE
Prevent skipping deathtiles with enough speed

### LINKED ISSUE #1936: deathtiles still skipable?
So in 0.6 it was possible to skip 1tile wide deathtiles with enough speed caused by either ninja sword, or enough grenade spam and gravity.

This is probably only relevant in Race but still it would be nice if it is fixed.

### PR DESCRIPTION
**Pull request by AssassinTee**
Fixes #1936 with MoveBox method

### CODE REVIEW

**Comment by AssassinTee** [comment]
That one filthy commit that won't go away

**Comment by Dune-jr** [comment]
For continuity, this follows PR https://github.com/teeworlds/teeworlds/pull/2113

File: src/game/collision.h [hunk 1]
```diff
@@ -1,5 +1,3 @@
-/* (c) Magnus Auvinen. See licence.txt in the root of the distribution for more information. */
```
Comment (oy) [inline src/game/collision.h]: Could you readd the copyright notice

File: src/game/collision.h [hunk 2]
```diff
@@ -25,15 +23,15 @@ class CCollision
...
+	bool CheckPoint(float x, float y, int Flag=COLFLAG_SOLID) const { return IsTile(round_to_int(x), round_to_int(y), Flag); }
+	bool CheckPoint(vec2 Pos, int Flag=COLFLAG_SOLID) const { return CheckPoint(Pos.x, Pos.y, Flag); }
 	int GetCollisionAt(float x, float y) const { return GetTile(round_to_int(x), round_to_int(y)); }
 	int GetWidth() const { return m_Width; };
 	int GetHeight() const { return m_Height; };
 	int IntersectLine(vec2 Pos0, vec2 Pos1, vec2 *pOutCollision, vec2 *pOutBeforeCollision) const;
 	void MovePoint(vec2 *pInoutPos, vec2 *pInoutVel, float Elasticity, int *pBounces) const;
-	void MoveBox(vec2 *pInoutPos, vec2 *pInoutVel, vec2 Size, float Elasticity) const;
-	bool TestBox(vec2 Pos, vec2 Size) const;
+	void MoveBox(vec2 *pInoutPos, vec2 *pInoutVel, vec2 Size, float Elasticity, bool *pDeath=NULL) const;
```
Comment (oy) [inline src/game/collision.h]: 0 instead of null

File: src/game/collision.h (hunk 1 above)
Comment (AssassinTee) [reply src/game/collision.h]: Huh, ofc

File: src/game/collision.h (hunk 2 above)
Comment (AssassinTee) [reply src/game/collision.h]: done

**Comment by Dune-jr** [comment]
This broke the nethash

**Pull request merged**